lucene/core/src/java/org/apache/lucene/search/comparators/DoubleComparator.java

```diff
@@ -59,6 +59,11 @@ protected long missingValueAsComparableLong() {
     return NumericUtils.doubleToSortableLong(missingValue);
   }
 
+  @Override
+  protected long sortableBytesToLong(byte[] bytes) {
+    return NumericUtils.sortableBytesToInt(bytes, 0);
```

Comment:
Hmm, should be `long` since doubles are 8-bytes?

```suggestion
    return NumericUtils.sortableBytesToLong(bytes, 0);
```